amps/ags/rm-automation/rm-automation-community-rest-api/src/test/java/org/alfresco/rest/rm/community/recordcategories/MoveRecCategoriesWithRSTests.java

```diff
@@ -0,0 +1,287 @@
+        dispositionScheduleService.addCutOffAfterEventStep(rootCategory2.getName(), RMEvents.OBSOLETE.getEventName());
+
+        STEP("Add retention schedule destroy step after 2 Day period.");
+        dispositionScheduleService.addDestroyWithGhostingAfterPeriodStep(rootCategory2.getName(), "day|2", DATE_FILED);
+
+        STEP("Move the subcategory within the rootCategory2.");
+        getRestAPIFactory().getNodeAPI(toContentModel(subCategory.getId())).move(createBodyForMoveCopy(rootCategory2.getId()));
+        assertStatusCode(OK);
+
+        STEP("Check that both records inherit rootCategory2 retention schedule");
+        elRecord = getRestAPIFactory().getRecordsAPI().getRecord(elRecord.getId());
+        nonElRecord = getRestAPIFactory().getRecordsAPI().getRecord(nonElRecord.getId());
+        assertTrue(elRecord.getProperties().getRecordSearchDispositionActionName().equalsIgnoreCase(RM_ACTIONS.CUT_OFF.getAction()),
+                "Disposition action should be cut off");
+        assertTrue(elRecord.getProperties().getRecordSearchDispositionPeriod().equalsIgnoreCase("none"),
+                "Disposition period property should be none");
+        assertTrue(elRecord.getProperties().getRecordSearchDispositionPeriodExpression().equalsIgnoreCase("0"),
+                "Disposition period expression should be 0");
+        assertFalse(elRecord.getProperties().getRecordSearchDispositionEvents().contains(RMEvents.CASE_CLOSED.getEventName()),
+                "Event list contain the event from the previous RS ");
+        assertTrue(elRecord.getProperties().getRecordSearchDispositionEvents().contains(RMEvents.OBSOLETE.getEventName()),
+                "Event list doesn't contain the event from the current RS ");
+        assertTrue(nonElRecord.getProperties().getRecordSearchDispositionActionName().equalsIgnoreCase(RM_ACTIONS.CUT_OFF.getAction()),
+                "Disposition action should be cut off");
+        assertTrue(nonElRecord.getProperties().getRecordSearchDispositionPeriod().equalsIgnoreCase("none"),
+                "Disposition period property should be none");
+        assertTrue(nonElRecord.getProperties().getRecordSearchDispositionPeriodExpression().equalsIgnoreCase("0"),
+                "Disposition period expression should be 0");
+        assertFalse(nonElRecord.getProperties().getRecordSearchDispositionEvents().contains(RMEvents.CASE_CLOSED.getEventName()),
+                "Event list contain the event from the previous RS ");
+        assertTrue(nonElRecord.getProperties().getRecordSearchDispositionEvents().contains(RMEvents.OBSOLETE.getEventName()),
+                "Event list doesn't contain the event from the current RS ");
+    }
+
+    @AfterClass (alwaysRun = true)
+    public void cleanupMoveRecCategoriesWithRSTests()
+    {
+        categoriesToBeDeleted.forEach(cat -> getRestAPIFactory().getRecordCategoryAPI().deleteRecordCategory(cat));
+    }
+}
```

Comment:
I would try to somehow remove some of the duplicated code. For example I would create both root categories in before method and delete them in after method and I would create a helper method for creating the  subcategory-record folder-records structure.

Reply:
Thanks for the suggestions! I have update the tests